### DISCUSSION TITLE
ScriptReference/Event-modifiers

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#189] [Notes]
# ScriptReference/Event-modifiers

https://docs.unity3d.com/ScriptReference/Event-modifiers.html

**Comment by mvi** [comment]
Note that this will not return `EventModifiers.None` when no keys are held if either Caps Lock or Num Lock is enabled. As these are easily left on by the user you should generally discount them from your comparisons.  
For example if you wish to check if they are holding shift (but not control, command etc), rather than do:  
  

```csharp
if (e.modifiers == EventModifiers.Shift)
```
  
Which would fail if the player accidentally has Caps Lock on, instead do something like:  
  

```csharp
EventModifiers modfiers = e.modifiers;  
modifiers &= (~EventModifiers.CapsLock);  
modifiers &= (~EventModifiers.Numeric);  
if (modifiers == EventModifiers.Shift)
```
